common/src/main/java/com/wynntils/wynn/model/container/ContainerQueryManager.java

```diff
@@ -58,24 +62,40 @@ public static void runQuery(ContainerQueryStep firstStep) {
 
         currentStep = firstStep;
         firstStepName = firstStep.getName();
-
+        lastStep = null;
+        resetTimer();
         if (!firstStep.startStep(null)) {
             raiseError("Cannot execute first step");
         }
     }
 
     @SubscribeEvent
     public static void onMenuOpened(MenuEvent.MenuOpenedEvent e) {
-        if (currentStep == null) return;
+        if (currentStep == null && lastStep == null) return;
+        if (currentStep == null && lastStep != null) {
```

Comment:
Can't we "simplify" this? Or did you want to avoid right shift?

Reply:
Different opinions on what "simple" means. The first check is the normal case, just wanted it to be clearly visible. But I agree it looks not so nice, will fix.